base_layer/wallet_ffi/src/lib.rs

```diff
@@ -4135,6 +4138,106 @@ pub unsafe extern "C" fn wallet_get_seed_words(wallet: *mut TariWallet, error_ou
     }
 }
 
+/// Gets the current emoji set
+///
+/// ## Arguments
+/// `()` - Does not take any arguments
+///
+/// ## Returns
+/// `*mut EmojiSet` - Pointer to the created EmojiSet.
+///
+/// # Safety
+/// The ```emoji_set_destroy``` function must be called when finished with a ByteVector to prevent a memory leak
+#[no_mangle]
+pub unsafe extern "C" fn get_emoji_set() -> *mut EmojiSet {
+    let current_emoji_vec = emoji_set();
+    let mut emoji_set: Vec<ByteVector> = Vec::new();
+    for emoji in current_emoji_vec {
+        let mut b = [0; 4]; // emojis are 4 bytes, unicode character
+        let emoji_char = ByteVector(emoji.encode_utf8(&mut b).as_bytes().to_vec());
+        emoji_set.push(emoji_char);
+    }
+    let result = EmojiSet(emoji_set);
+    Box::into_raw(Box::new(result))
+}
+
+/// Gets the length of the current emoji set
+///
+/// ## Arguments
+/// `*mut EmojiSet` - Pointer to emoji set
+///
+/// ## Returns
+/// `c_int` - Pointer to the created EmojiSet.
+///
+/// # Safety
+/// None
+#[no_mangle]
+pub unsafe extern "C" fn emoji_set_get_length(emoji_set: *const EmojiSet, error_out: *mut c_int) -> c_uint {
+    let mut error = 0;
+    ptr::swap(error_out, &mut error as *mut c_int);
+    if emoji_set.is_null() {
+        error = LibWalletError::from(InterfaceError::NullError("emoji_set".to_string())).code;
+        ptr::swap(error_out, &mut error as *mut c_int);
+        return 0;
+    }
+    (*emoji_set).0.len() as c_uint
+}
+
+/// Gets a ByteVector at position in a EmojiSet
+///
+/// ## Arguments
+/// `emoji_set` - The pointer to a EmojiSet
+/// `position` - The integer position
+/// `error_out` - Pointer to an int which will be modified to an error code should one occur, may not be null. Functions
+/// as an out parameter.
+///
+/// ## Returns
+/// `ByteVector` - Returns a ByteVector. Note that the ByteVector will be null if ptr
+/// is null or if the position is invalid
+///
+/// # Safety
+/// The ```byte_vector_destroy``` function must be called when finished with the ByteVector to prevent a memory leak.
+#[no_mangle]
+pub unsafe extern "C" fn emoji_set_get_at(
+    emoji_set: *const EmojiSet,
+    position: c_uint,
+    error_out: *mut c_int,
+) -> *mut ByteVector
+{
+    let mut error = 0;
+    ptr::swap(error_out, &mut error as *mut c_int);
+    if emoji_set.is_null() {
+        error = LibWalletError::from(InterfaceError::NullError("emoji_set".to_string())).code;
+        ptr::swap(error_out, &mut error as *mut c_int);
+        return ptr::null_mut();
+    }
+    let len = emoji_set_get_length(emoji_set, error_out) as c_uint - 1;
+    if position > len as c_uint {
```

Comment:
Some unnecessary type coercions. `len` is not a length it is the index of the last item (renamed to `last_index`).

```suggestion
    let last_index = emoji_set_get_length(emoji_set, error_out) - 1;
    if position > last_index {
```